docs/tutorials/hub/web-app.md

```diff
@@ -0,0 +1,85 @@
+
+## Getting Started
+
+There are a few resources you'll need before you start writing code.
+
+- [An account](../hub/accounts.md). This is your developer account on the Hub.
+- [A user group key](../hub/app-apis.md). This is how your users will be able to access your Hub APIs. Consider creating the key in an organization not your personal account so that you can invite collaborators later.
+- [A new Typescript web app](https://webpack.js.org/guides/typescript/). We recommend using Typescript, as Textile libraries are in a stage rapid of development and type detection is valuable during upgrades.
+
+### API Access
+
+A [_user group key_](../hub/app-apis.md) consists of a _key_ and a _secret_, which you'll never want to expose to users. However, for users to access your Hub resources, they'll need a token derived from their own identity and the user group's key and secret. So to use the Hub APIs in your web app, you have two options:
+
+* Build a light-weight token endpoint accessible only by your app ([below](#creating-a-token-endpoint)).
+* Use the domain whitelisting to skip token generation ([in development](https://github.com/textileio/textile/issues/109))
+
+We'll come back to option one in the second part of this tutorial.
+
+## Building a simple web app
+
+With your basic Typescript web app setup, you should have a primary `.ts` (or `.js` if you decided to stick with Javascript) file where your app is written. We'll work within this single file to start, but the instructions below should be adaptable to any application architecture.
+
+### Generating an Identity
+
+For identity in our app, we'll use a private-key based identity generated randomly for each new app user. To create a function for creating and restoring identies, we can use the _cached identity example_ from the [basic libp2p identities tutorial](libp2p-identities.md).
+
+```typescript
+const identity = getIdentity();
+```
+
+!!!info
+    In the future we will support email-based identities, allowing you to more easily use the Hub APIs with common email-based user models. [Follow progress here](https://github.com/textileio/textile/issues/216).
+
+### Creating a token endpoint
+
+API tokens need to be generated for every new user. Additionally, well designed tokens expire quickly, so new tokens will need to be generated for user's each time their token expires. There is two ways to handle token generate. The first is using the `getToken` method and the second is using the `getTokenChallenge`, both available in `@textile/threads-client`. Both of those methods require Hub API key and secret, so should be executed by a service only accessible to your app.
+
+**When to use the getToken method**
+
+The `getToken` method should only be used in rare cases where the application has access to the user's private key _outside of the app_. Examples might include cases where private key identities are generated server-side.
```

Comment:
In reality, this is probably overly restrictive. This method doesn't assume you have access to the key either, just that the object you are passing has a 'sign' method, which could actually abstract exactly the same way as the getTokenChallenge method. In fact, getToken simply wraps getTokenChallenge. With that in mind, I do get the need/want to have a reason for the two methods, but maybe with tempered language here?

Reply:
I think that's only true if you ran those methods directly from an app. Which you cannot do in this example, so here I'm referring to server-side token gen, in which case i don't think that's true. Or am I mistaken?